presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java

```diff
@@ -62,12 +63,12 @@
     private final LocalMemoryContext pageSourceMemoryContext;
     private final LocalMemoryContext pageBuilderMemoryContext;
     private final SettableFuture<?> blocked = SettableFuture.create();
+    private final MergingPageOutput currentOutput;
```

Comment:
`currentOutput` doesn't change through lifetime of the operator (so it's always current and the name is meaningless). Consider renaming it to `mergingOutput`.